### PR TITLE
Improve error handling for RG deployment validation:

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

Fix for a couple of the issues outlined in issue: https://github.com/Azure/azure-powershell/issues/6856

- Unwrap nested exceptions thrown outside of validation
- Add -IgnoreDynamicParameters switch to enable some automation scenarios

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Resources/Resources/help/Test-AzResourceGroupDeployment.md [hunk 1]
```diff
@@ -144,6 +153,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -IgnoreDynamicParameters
+Skip resolving dynamic parameters; this should be used when running a script or running this command as a job.
```
Comment (markcowl) [inline src/Resources/Resources/help/Test-AzResourceGroupDeployment.md]: Not quite.  This should be used if all template parameters are satisfied using a templateparamaterobject or templateparameterfile

File: src/Resources/ResourceManager/Implementation/CmdletBase/ResourceWithParameterCmdletBase.cs [hunk 2]
```diff
@@ -116,69 +116,75 @@ protected ResourceWithParameterCmdletBase()
         [ValidateNotNullOrEmpty]
         public string TemplateUri { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Skip resolving dynamic parameters; this should be used when running a script or running this command as a job.")]
+        public SwitchParameter IgnoreDynamicParameters { get; set; }
```
[the remark this answers is not in the document]
Comment (markcowl) [reply src/Resources/ResourceManager/Implementation/CmdletBase/ResourceWithParameterCmdletBase.cs]: Aso, isn't the actually right thing to do to only provide these checks if no TemplateParameterObject or TemplateParameterFiel are provided?

That is, we only prompt for template parameters if the user doesn't provide parameters some other way.  I realize this is a breaking change, but this seems like the most intuitive option.  Generally if I am providing one of these objects I am going to think that I have satisfied all of the template parameters.

[the remark this answers is not in the document]
Comment (cormacpayne) [reply src/Resources/ResourceManager/Implementation/CmdletBase/ResourceWithParameterCmdletBase.cs]: Talked offline -- changed the name to `-SkipTemplateParameterPrompt` and updated description to properly reflect this parameter's function.

File: src/Resources/Resources/help/Test-AzResourceGroupDeployment.md (hunk 1 above)
Comment (cormacpayne) [reply src/Resources/Resources/help/Test-AzResourceGroupDeployment.md]: Same comment -- updated parameter name and description.

File: src/Resources/ResourceManager/Implementation/CmdletBase/ResourceWithParameterCmdletBase.cs [hunk 3]
```diff
@@ -116,69 +116,78 @@ protected ResourceWithParameterCmdletBase()
         [ValidateNotNullOrEmpty]
         public string TemplateUri { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Skips the PowerShell dynamic parameter processing that checks if the provided template parameter contains all necessary parameters used by the template. " +
+                                                    "This check would prompt the user to provide a value for the missing parameters, but providing the -SkipTemplateParameterPrompt will ignore this prompt and " +
+                                                    "error out immediately if a parameter was found not to be bound in the template. For non-interactive scripts, -SkipTemplateParameterPrompt can be provided " +
+                                                    "to avoid generic error messages from being thrown.")]
```
Comment (markcowl) [inline src/Resources/ResourceManager/Implementation/CmdletBase/ResourceWithParameterCmdletBase.cs]: To provide a better error message in the case where not all required parameters are satisfied

File: tools/StaticAnalysis/Exceptions/Az.Resources/SignatureIssues.csv [hunk 4]
```diff
@@ -26,4 +26,4 @@
...
 "Microsoft.Azure.PowerShell.Cmdlets.ResourceManager.Cmdlets.dll","Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.SetAzureResourceLockCmdlet","Set-AzResourceLock","1","8410","Parameter LockNotes of cmdlet Set-AzResourceLock does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
-"Microsoft.Azure.PowerShell.Cmdlets.ResourceManager.Cmdlets.dll","Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.GetAzureResourceCmdlet","Get-AzResource","1","8410","Parameter ExpandProperties of cmdlet Get-AzResource does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
+"Microsoft.Azure.PowerShell.Cmdlets.ResourceManager.Cmdlets.dll","Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.GetAzureResourceCmdlet","Get-AzResource","1","8410","Parameter ExpandProperties of cmdlet Get-AzResource does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
```
Comment (markcowl) [inline tools/StaticAnalysis/Exceptions/Az.Resources/SignatureIssues.csv]: revert this change
